<?xml version="1.0" encoding="UTF-8"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro" name="UMIUSI">

  <xacro:macro name="sinsei_umiusi_control"
    params="
      thruster_mode
      high_beam_pin
      low_beam_pin
      ir_pin
      indicator_led_pin
      esc_direct1_pin
      esc_direct2_pin
      esc_direct3_pin
      esc_direct4_pin
      servo_direct1_pin
      servo_direct2_pin
      servo_direct3_pin
      servo_direct4_pin
      esc_direct1_center_pulse_width
      esc_direct1_negative_pulse_width_radius
      esc_direct1_positive_pulse_width_radius
      esc_direct2_center_pulse_width
      esc_direct2_negative_pulse_width_radius
      esc_direct2_positive_pulse_width_radius
      esc_direct3_center_pulse_width
      esc_direct3_negative_pulse_width_radius
      esc_direct3_positive_pulse_width_radius
      esc_direct4_center_pulse_width
      esc_direct4_negative_pulse_width_radius
      esc_direct4_positive_pulse_width_radius
      servo_direct1_min_pulse_width
      servo_direct1_max_pulse_width
      servo_direct2_min_pulse_width
      servo_direct2_max_pulse_width
      servo_direct3_min_pulse_width
      servo_direct3_max_pulse_width
      servo_direct4_min_pulse_width
      servo_direct4_max_pulse_width
      vesc1_id
      vesc2_id
      vesc3_id
      vesc4_id
      period_led_tape_per_thrusters
      mock
    ">

    <xacro:include filename="sinsei_umiusi_control/can.urdf.xacro" />
    <xacro:include filename="sinsei_umiusi_control/thruster_direct.urdf.xacro" />
    <xacro:include filename="sinsei_umiusi_control/indicator_led.urdf.xacro" />
    <xacro:include filename="sinsei_umiusi_control/imu.urdf.xacro" />
    <xacro:include filename="sinsei_umiusi_control/headlights.urdf.xacro" />

    <xacro:can
      thruster_mode="${thruster_mode}"
      vesc1_id="${vesc1_id}"
      vesc2_id="${vesc2_id}"
      vesc3_id="${vesc3_id}"
      vesc4_id="${vesc4_id}"
      period_led_tape_per_thrusters="${period_led_tape_per_thrusters}"
    />
    <xacro:if value="${thruster_mode == 'direct'}">
      <xacro:thruster_direct
        esc1_pin="${esc_direct1_pin}"
        esc2_pin="${esc_direct2_pin}"
        esc3_pin="${esc_direct3_pin}"
        esc4_pin="${esc_direct4_pin}"
        servo1_pin="${servo_direct1_pin}"
        servo2_pin="${servo_direct2_pin}"
        servo3_pin="${servo_direct3_pin}"
        servo4_pin="${servo_direct4_pin}"
        esc1_center_pulse_width="${esc_direct1_center_pulse_width}"
        esc1_negative_pulse_width_radius="${esc_direct1_negative_pulse_width_radius}"
        esc1_positive_pulse_width_radius="${esc_direct1_positive_pulse_width_radius}"
        esc2_center_pulse_width="${esc_direct2_center_pulse_width}"
        esc2_negative_pulse_width_radius="${esc_direct2_negative_pulse_width_radius}"
        esc2_positive_pulse_width_radius="${esc_direct2_positive_pulse_width_radius}"
        esc3_center_pulse_width="${esc_direct3_center_pulse_width}"
        esc3_negative_pulse_width_radius="${esc_direct3_negative_pulse_width_radius}"
        esc3_positive_pulse_width_radius="${esc_direct3_positive_pulse_width_radius}"
        esc4_center_pulse_width="${esc_direct4_center_pulse_width}"
        esc4_negative_pulse_width_radius="${esc_direct4_negative_pulse_width_radius}"
        esc4_positive_pulse_width_radius="${esc_direct4_positive_pulse_width_radius}"
        servo1_min_pulse_width="${servo_direct1_min_pulse_width}"
        servo1_max_pulse_width="${servo_direct1_max_pulse_width}"
        servo2_min_pulse_width="${servo_direct2_min_pulse_width}"
        servo2_max_pulse_width="${servo_direct2_max_pulse_width}"
        servo3_min_pulse_width="${servo_direct3_min_pulse_width}"
        servo3_max_pulse_width="${servo_direct3_max_pulse_width}"
        servo4_min_pulse_width="${servo_direct4_min_pulse_width}"
        servo4_max_pulse_width="${servo_direct4_max_pulse_width}"
      />
    </xacro:if>
    <xacro:imu />
    <xacro:headlights
      high_beam_pin="${high_beam_pin}"
      low_beam_pin="${low_beam_pin}"
      ir_pin="${ir_pin}" />
    <xacro:indicator_led
      pin="${indicator_led_pin}"
    />
  </xacro:macro>

</robot>
